cmake/gcovr.cmake

```diff
@@ -0,0 +1,16 @@
+if(USE_GCOV)
+ 
+    SET(GCOVR gcovr)
+ 
+    SET(GCC_COVERAGE_COMPILE_FLAGS "-fprofile-arcs -ftest-coverage")
+    SET(GCC_COVERAGE_LINK_FLAGS "-lgcov")
+    SET( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${GCC_COVERAGE_COMPILE_FLAGS}" )
+    SET( CMAKE_EXE_LINKER_FLAGS "${CMAKE_LINKER_FLAGS} ${GCC_COVERAGE_LINK_FLAGS}" )
+             
+    file(MAKE_DIRECTORY ${CMAKE_BINARY_DIR}/coverage_output )
+    add_custom_command(OUTPUT _run_gcovr_parser
+    POST_BUILD
+    COMMAND ${GCOVR} -r ${CMAKE_SOURCE_DIR}/src --object-dir=${CMAKE_BINARY_DIR} --html --html-details -o coverage_output/coverage.html
```

Comment:
Two comments here:

1. Please, use the long options instead of the short one for improving readability of the code: `--root` instead of `-r`. 
2. I would remote the `/src` part from the ROOT component. I would be interested in seeing the coverage in the full project (public headers , apps & samples included) instead of only obtaining the coverage in the src folder.